examples/collector/README.md

```diff
@@ -0,0 +1,145 @@
+
+
+
+## Starting a plugin
+
+After implementing a type that satisfies one of {collector, processor, publisher} interfaces, all that is left to do is a call the appropriate plugin.StartX() with your plugin specific options. That could be as simple as:
+
+```go
+	plugin.StartCollector(rand.RandCollector{}, pluginName, pluginVersion)
+```
+
+### Meta options
+
+The available options are defined in [plugin/meta.go](https://github.com/intelsdi-x/snap-plugin-lib-go/tree/master/v/1/plugin/meta.go). You can use some or none of the options. The options with definitions/explanations are below:
+
+```go
+// ConcurrencyCount is the max number of concurrent calls the plugin
+// should take.  For example:
+// If there are 5 tasks using the plugin and its concurrency count is 2,
+// snapd will keep 3 plugin instances running.
+// ConcurrencyCount overwrites the default (5) for a Meta's ConcurrencyCount.
+func ConcurrencyCount(cc int) MetaOpt {
+}
+
+// Exclusive == true results in a single instance of the plugin running
+// regardless of the number of tasks using the plugin.
+// Exclusive overwrites the default (false) for a Meta's Exclusive key.
+func Exclusive(e bool) MetaOpt {
+}
+
+// Unsecure results in unencrypted communication with this plugin.
+// Unsecure overwrites the default (false) for a Meta's Unsecure key.
+func Unsecure(e bool) MetaOpt {
+}
+
+// RoutingStrategy will override the routing strategy this plugin requires.
+// The default routing strategy is Least Recently Used.
+// RoutingStrategy overwrites the default (LRU) for a Meta's RoutingStrategy.
+func RoutingStrategy(r router) MetaOpt {
+}
+
+// CacheTTL will override the default cache TTL for the this plugin. snapd
+// caches metrics on the daemon side for a default of 500ms.
+// CacheTTL overwrites the default (500ms) for a Meta's CacheTTL.
+func CacheTTL(t time.Duration) MetaOpt {
+}
+```
+
+An example of what using all of them would look like:
+
+```go
+	plugin.StartCollector(mypackage.Mytype{},
+				pluginName,
+				pluginVersion,
+				plugin.ConcurrencyCount(a),
+				plugin.Exclusive(b),
+				plugin.Unsecure(c),
+				plugin.RoutingStrategy(d),
+				plugin.CacheTTL(e))
+```
+
+## Testing
+For testing reference the [Snap Testing Guidelines](https://github.com/intelsdi-x/snap/blob/master/CONTRIBUTING.md#testing-guidelines). To test your plugin with Snap you will need to have [Snap](https://github.com/intelsdi-x/snap) installed, check out these docs for [Snap setup details](https://github.com/intelsdi-x/snap/blob/master/docs/BUILD_AND_TEST.md#getting-started).
+
+You will have to specify the [test size](https://github.com/intelsdi-x/snap/blob/master/docs/BUILD_AND_TEST.md#test) (either small, medium, or large) with the specified build tag `// +build test-size`. The default testing will run all size tests if the build tag is not specified.
```

Comment:
Each test file should specify the appropriate build tag such as "small", "medium", "large" (e.g. `// +build small`).

~~The default testing will run all size tests if the build tag is not specified.~~  If we leave this in it should include how `go test` should also be run with the option `-tags small` if the intent was to only run small tests.